### PR TITLE
test(utils): add boundary tests for Older::check_older and After::check_after

### PR DESCRIPTION
**Pull request by KrishnaShuk**
### Description

Addresses the `// TODO: test >= / >` on line 111 of [utils.rs](cci:7://file:///home/krishna/Contribution/bdk_wallet/src/test_utils.rs:0:0-0:0). Adds 11 boundary-condition tests for the `Satisfier` impls of [Older](cci:2://file:///home/krishna/Contribution/bdk_wallet/src/wallet/utils.rs:87:0-91:1) and [After](cci:2://file:///home/krishna/Contribution/bdk_wallet/src/wallet/utils.rs:43:0-46:1).

### Notes to the reviewers

Uses `Satisfier::<PublicKey>::check_older(...)` fully-qualified syntax because the trait is generic over `Pk`.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `just p` before pushing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

File: src/wallet/utils.rs [hunk 1]
```diff
@@ -278,4 +278,116 @@ mod test {
...
+
+    // --- After::check_after boundary tests ---
+
+    #[test]
+    fn test_check_after_at_boundary() {
+        // current_height == locktime → should pass (>=)
+        let after = After::new(Some(100_000), false);
+        assert!(Satisfier::<PublicKey>::check_after(
+            &after,
+            absolute::LockTime::from_consensus(100_000)
```
Comment (busayo-OD) [inline src/wallet/utils.rs:354]: `absolute::LockTime` can also represent Unix timestamps (>= 500_000_000), but `After` only has `current_height` and no `current_time`. So a timestamp-based locktime would end up being compared against a block height. I also noticed the existing tests don't cover this. Is this intentionally limited to height-based locktimes for now?

Comment (busayo-OD) [reply src/wallet/utils.rs:354]: I traced the satisfaction flow further. Timestamp CLTV finalization works; existing wallet tests cover that path, and PsbtInputSatisfier::check_after() evaluates the transaction's lock_time via is_implied_by, which is locktime-type aware.

One thing I noticed: After::check_after() does current_height >= n.to_consensus_u32(), which doesn't distinguish between height- and timestamp-based locktimes. Existing timestamp CLTV tests still pass through the signing/finalization flow.
